### PR TITLE
feat(core/dropdownbutton): migrate to shadow dom

### PR DESCRIPTION
**Pull request by danielleroux**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [x] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: #511 

## What is the new behavior / Open Points?

- [x] Migrated dropdown-button to shadow-dom
- [x] Refactored ix-dropdown-item
- [x] Add ix-dropdown-header
- [x] Removed old `Placements` (See BRAKING_CHANGES.md)
- [x] Fix sub menu behavior
- [ ] Clarify padding in spec of dropdown-item
- [ ] Fix failing vrt

## Does this introduce a breaking change?

- [x] Yes, see BREAKING_CHANGES.md
- [ ] No

### CODE REVIEW

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
 ## v2.0.0
 
+### `ix-dropdown` removed deprecated properties
+
+- adjustDropdownWidthToReferenceWith
+- adjustDropdownWidthToReferenceWidth
+
+### `ix-dropdown` submenu
+
+Submenu append chevron icon automatically to the trigger element, if the trigger element is an ix-dropdown-item.
```
Comment (nuke-ellington) [inline BREAKING_CHANGES.md:12]:
```suggestion
The submenu now appends a chevron icon automatically to the trigger element, if the trigger element is an ix-dropdown-item.
```

File: packages/core/src/components/dropdown-header/dropdown-header.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  padding: 0.25rem 0.5rem;
+  padding-left: 2rem;
+
+  overflow: hidden;
+  cursor: default;
+
+  color: var(--theme-color-soft-text);
+
+  .category-text {
+  }
```
Comment (nuke-ellington) [inline packages/core/src/components/dropdown-header/dropdown-header.scss:29]:
```suggestion
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/tests/select/select.e2e.ts-snapshots/select-autoplacement-1-chromium---theme-classic-dark-linux.png
Comment (nuke-ellington) [inline packages/core/src/tests/select/select.e2e.ts-snapshots/select-autoplacement-1-chromium---theme-classic-dark-linux.png:1]: Is the left margin of the checkmark correct? Looks very narrow :)

Comment (danielleroux) [reply packages/core/src/tests/select/select.e2e.ts-snapshots/select-autoplacement-1-chromium---theme-classic-dark-linux.png:1]: Updated please check

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**